File: app/src/main/java/com/phoenix/phoenix/entity/Programmazione.java
package com.phoenix.phoenix.entity;

import com.phoenix.phoenix.converter.PostoConverter;
import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Collection;
import java.util.Date;

@Data
@NoArgsConstructor
@Entity
@Table(name = "programmazione")
public class Programmazione {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @OneToOne
    private Pellicola pellicola;

    @OneToOne
    private Sala sala;

    @Convert(converter = PostoConverter.class)
    private Collection<Posto> posti;

    private double prezzo;

    private Date orario;

}

File: front/src/app/model/programmazione.ts
import { Pellicola } from "./pellicola";
import { Posto } from "./posto";
import { Sala } from "./sala";

export interface Programmazione{
    id:number;
    pellicola:Pellicola;
    sala:Sala;
    posti:Array<Posto>;
    prezzo:number;
    orario:Date;
}